rest-api/dao/metrics_cache_dao.py

```diff
@@ -0,0 +1,106 @@
+from model.metrics_cache import MetricsEnrollmentStatusCache, MetricsGenderCache
+from dao.base_dao import BaseDao
+import datetime
+
+class MetricsEnrollmentStatusCacheDao(BaseDao):
+  def __init__(self):
+    super(MetricsEnrollmentStatusCacheDao, self).__init__(MetricsEnrollmentStatusCache)
+
+  def get_serving_version_with_session(self, session):
+    return (session.query(MetricsEnrollmentStatusCache)
+            .order_by(MetricsEnrollmentStatusCache.dateInserted.desc())
+            .first())
+
+  def get_active_buckets(self, start_date=None, end_date=None):
+    with self.session() as session:
+      last_inserted_record = self.get_serving_version_with_session(session)
+      if last_inserted_record is None:
+        return None
+      last_inserted_date = last_inserted_record.dateInserted
+      query = session.query(MetricsEnrollmentStatusCache)\
+        .filter(MetricsEnrollmentStatusCache.dateInserted == last_inserted_date)
+      if start_date:
+        query = query.filter(MetricsEnrollmentStatusCache.date >= start_date)
+      if end_date:
+        query = query.filter(MetricsEnrollmentStatusCache.date <= end_date)
+      return query.order_by(MetricsEnrollmentStatusCache.date)\
+        .order_by(MetricsEnrollmentStatusCache.hpoId).all()
+
+  def delete_old_records(self, n_days_ago=7):
+    with self.session() as session:
+      last_inserted_record = self.get_serving_version_with_session(session)
+      if last_inserted_record is not None:
+        last_date_inserted = last_inserted_record.dateInserted
+        seven_days_ago = last_date_inserted - datetime.timedelta(days=n_days_ago)
+        delete_sql = """
+          delete from metrics_enrollment_status_cache where date_inserted < :seven_days_ago
```

Comment:
Should this be greater than?

Reply:
No, it's less than. This SQL is used to delete the old records before seven days timestamp to keep the table small.

Reply:
Ok it just reads confusing. “Less than seven days ago “ sounds earlier than.